bin/mixer_codegen.sh

```diff
@@ -99,27 +99,41 @@ fi
 
 # TODO: vendor those too like istio.io/api without pruning protos
 
+tmpdir=$(dirname $(mktemp -u))/protos_for_mixer_codegen
+
 GOOGLEAPIS_SHA=c8c975543a134177cc41b64cbbf10b88fe66aa1d
 GOOGLEAPIS_URL=https://raw.githubusercontent.com/googleapis/googleapis/${GOOGLEAPIS_SHA}
 
-if [ ! -e ${ROOT}/vendor/github.com/googleapis/googleapis ]; then
+if [ ! -e ${tmpdir}/googleapis/googleapis/google/rpc ]; then
 echo "Pull down source protos from googleapis..."
 
-mkdir -p ${ROOT}/vendor/github.com/googleapis/googleapis
+mkdir -p ${tmpdir}/googleapis/googleapis/google/rpc
 
 # all the google_rpc protos
-mkdir -p ${ROOT}/vendor/github.com/googleapis/googleapis/google/rpc
-curl -sS ${GOOGLEAPIS_URL}/google/rpc/status.proto > ${ROOT}/vendor/github.com/googleapis/googleapis/google/rpc/status.proto
-curl -sS ${GOOGLEAPIS_URL}/google/rpc/code.proto > ${ROOT}/vendor/github.com/googleapis/googleapis/google/rpc/code.proto
-curl -sS ${GOOGLEAPIS_URL}/google/rpc/error_details.proto > ${ROOT}/vendor/github.com/googleapis/googleapis/google/rpc/error_details.proto
+curl -sS ${GOOGLEAPIS_URL}/google/rpc/status.proto > ${tmpdir}/googleapis/googleapis/google/rpc/status.proto
+curl -sS ${GOOGLEAPIS_URL}/google/rpc/code.proto > ${tmpdir}/googleapis/googleapis/google/rpc/code.proto
+curl -sS ${GOOGLEAPIS_URL}/google/rpc/error_details.proto > ${tmpdir}/googleapis/googleapis/google/rpc/error_details.proto
+curl -sS ${GOOGLEAPIS_URL}/google/rpc/error_details.proto > ${tmpdir}/googleapis/googleapis/google/rpc/error_details.proto
+fi
+
+# feb 23rd of gogo sha
+GOGO_SHA=d5bc08a2c4c572e9bcff820aa125fe38ac3a3a42
+GOGO_URL=https://raw.githubusercontent.com/gogo/protobuf/${GOGO_SHA}
+
+if [ ! -e ${tmpdir}/gogo/protobuf/gogoproto/gogo.proto ]; then
+echo "Pull down source protos from gogo/protobuf/gogoproto/gogo.proto..."
+
+mkdir -p ${tmpdir}/gogo/protobuf/gogoproto/
+
+# all the gogoproto protos
+curl -sS ${GOGO_URL}/gogoproto/gogo.proto > ${tmpdir}/gogo/protobuf/gogoproto/gogo.proto
 fi
 
 imports=(
  "${ROOT}"
  "${ROOT}/vendor/istio.io/api"
- "${ROOT}/vendor/github.com/gogo/protobuf"
- "${ROOT}/vendor/github.com/gogo/protobuf/protobuf"
- "${ROOT}/vendor/github.com/googleapis/googleapis"
+ "${tmpdir}/gogo/protobuf"
+ "${tmpdir}/googleapis/googleapis"
```

Comment:
let's add what is missing ? (assuming it's a few files, which it is if I read the curl right)